plugins/speech/service.js

```diff
@@ -1,99 +1,104 @@
 const {ipcRenderer} = require('electron');
 
 (function () {
-	'use strict';
+    'use strict';
 
-	function SpeechService($rootScope, $translate) {
-		var service = {}
-		var callbacks = {}
-		var commandList = []
+    function SpeechService($rootScope, $translate) {
+        var service = {}
+        var callbacks = {}
+        var commandList = []
 
-		service.init = function (cb) {
+        service.init = function (cb) {
             // workaround so we can trigger requests at any time 
-			annyang.isListening = () => { return true }
+            annyang.isListening = () => { return true }
             // Set lenguage and debug state
-			annyang.setLanguage((typeof config.general.language != 'undefined') ? config.general.language : 'en-US')
-			annyang.debug(false)
+            annyang.setLanguage((typeof config.general.language != 'undefined') ? config.general.language : 'en-US')
+            annyang.debug(false)
 
             // add specified callback functions
-			if (isCallback(cb.listening)) {
-				callbacks.listening = function (bool) {
-					$rootScope.$apply(cb.listening(bool))
+            if (isCallback(cb.listening)) {
+                callbacks.listening = function (bool) {
+                    $rootScope.$apply(cb.listening(bool))
+                }
+            }
+            if (isCallback(cb.partialResult)) {
+                callbacks.partialResult = function (data) {
+                    $rootScope.$apply(cb.partialResult(data))
+                }
+            }
+            if (isCallback(cb.finalResult)) {
+                callbacks.finalResult = function (data) {
+                    $rootScope.$apply(cb.finalResult(data))
+                }
+            }
+            if (isCallback(cb.error)) {
+                callbacks.error = function (data) {
+                    $rootScope.$apply(cb.error(data))
+                }
+            }
+
+            ipcRenderer.on('hotword', () => {
+                callbacks.listening(true)
+            })
+
+            ipcRenderer.on('partial-results', (event, text) => {
+                callbacks.partialResult(text)
+            })
+
+            ipcRenderer.on('final-results', (event, text) => {
+                callbacks.finalResult(text)
+                annyang.trigger(text)
+                callbacks.listening(false)
+            })
+
+        }
+
+				service.cleanup = function(){
```

Comment:
the problem with timeout, is that it is just the UI..  we have an issue, #592 which has the UI and the detached sonus task out of synch.   the reset should REALLY get the voice listener reset too.. just don't know how to do that.. (kill and restart?), so the reset doesn't really work..